### ISSUE TITLE
Support providing items for sidebar and main in Template Constructor

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#1693]
I believe it's a natural thing to provide one item or a list items to the Template constructor in the header, sidebar or main. But it is currently not supported

```python
from panel import template
from panel.pane import Markdown

ABOUT = """
# About the Awesome Panel Project
"""

def view() -> template.BaseTemplate:
    """The about view of awesome-panel.org"""
    main = [Markdown(ABOUT, sizing_mode="stretch_width")]
    return template.GoldenTemplate(title="About", main=main)

if __name__.startswith("bokeh"):
    view().servable()
```

raises

```bash
$ 2020-10-25 04:19:49,381 Starting Bokeh server version 2.2.3 (running on Tornado 6.0.4)
2020-10-25 04:19:49,383 User authentication hooks NOT provided (default user enabled)
2020-10-25 04:19:49,386 Bokeh app running at: http://localhost:5006/issue_template
2020-10-25 04:19:49,386 Starting Bokeh server with process id: 2500
2020-10-25 04:21:35,461 Error running application handler <bokeh.application.handlers.script.ScriptHandler object at 0x00000197CD7090C8>: Parameter 'main' value must be an instance of ListLike, not '[Markdown(str, sizing_mode='stretch_width')]'
File "__init__.py", line 1320, in _validate:
(self.name, class_name, val)) Traceback (most recent call last):
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\bokeh\application\handlers\code_runner.py", line 197, in run
    exec(self._code, module.__dict__)
  File "C:\repos\private\awesome-panel\scripts\issue_template.py", line 37, in <module>
  File "C:\repos\private\awesome-panel\scripts\issue_template.py", line 34, in view
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\panel\template\base.py", line 409, in __init__
    super(BasicTemplate, self).__init__(template=template, **params)
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\panel\template\base.py", line 61, in __init__
    super(BaseTemplate, self).__init__(**params)
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\param\parameterized.py", line 2339, in __init__
    self.param._setup_params(**params)
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\param\parameterized.py", line 991, in override_initialization
    fn(parameterized_instance,*args,**kw)
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\param\parameterized.py", line 1202, in _setup_params
    setattr(self,name,val)
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\param\parameterized.py", line 296, in _f
    return f(self, obj, val)
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\param\parameterized.py", line 823, in __set__
    self._validate(val)
  File "C:\repos\private\awesome-panel\.venv\lib\site-packages\param\__init__.py", line 1320, in _validate
    (self.name, class_name, val))
ValueError: Parameter 'main' value must be an instance of ListLike, not '[Markdown(str, sizing_mode='stretch_width')]'
```

**Comment by MarcSkovMadsen** [comment]
I started working on it. I can see that there is the issue that not all templates has `ListLike` header, sidebar and main items. For example ReactTemplate has not.

But that is sort of another issue that if you want to support multiple templates like for example in `test_manual.py` or as I would like to do at awesome-panel.org then there is no generic way to construct a Template with just a list of items.

You will have to write special code like

![image](https://user-images.githubusercontent.com/42288570/97098301-acc03c00-167b-11eb-9794-a4ae52c9f851.png)

To handle the ReactTemplate.

**Comment by philippjfr** [comment]
Closed in #1694
